kernel/include/kernel_internal.h

```diff
@@ -46,9 +46,21 @@ extern void z_setup_new_thread(struct k_thread *new_thread,
 			      void *p1, void *p2, void *p3,
 			      int prio, u32_t options, const char *name);
 
-extern void z_new_thread_init(struct k_thread *thread,
-					    char *pStack, size_t stackSize,
-					    int prio, unsigned int options);
+static inline void z_new_thread_init(struct k_thread *thread,
+				     char *stack, size_t stack_size)
+{
+#if !defined(CONFIG_INIT_STACKS) && !defined(CONFIG_THREAD_STACK_INFO)
+	ARG_UNUSED(stack);
+	ARG_UNUSED(stack_size);
+#endif
+#ifdef CONFIG_INIT_STACKS
+	memset(stack, 0xaa, stack_size);
+#endif
+#if defined(CONFIG_THREAD_STACK_INFO)
+	thread->stack_info.start = (uintptr_t)stack;
+	thread->stack_info.size = (u32_t)stack_size;
```

Comment:
This should be `thread->stack_info.size = (size_t)stack_size;`

Reply:
fixed.